tests/upgrades/test_activation_key.py

```diff
@@ -21,109 +21,114 @@
-    ).create()
-    custom_repo2.sync()
-    cv2 = entities.ContentView(organization=org[0], repository=[custom_repo2.id]).create()
-    cv2.publish()
-    org_subscriptions = entities.Subscription(organization=org[0]).search()
-    for subscription in org_subscriptions:
-        provided_products_ids = [prod.id for prod in subscription.read().provided_product]
-        if custom_repo2.product.id in provided_products_ids:
-            ak[0].add_subscriptions(data={'quantity': 1, 'subscription_id': subscription.id})
-    ak_subscriptions = ak[0].product_content()['results']
-    assert custom_repo2.product.id in {subscr['product']['id'] for subscr in ak_subscriptions}
-    ak[0].delete()
-    with pytest.raises(HTTPError):
-        entities.ActivationKey(id=ak[0].id).read()
-    for cv_object in [cv2, cv[0]]:
-        cv_contents = cv_object.read_json()
-        cv_object.delete_from_environment(cv_contents['environments'][0]['id'])
-        cv_object.delete(cv_object.organization.id)
-    custom_repo2.delete()
-    org[0].delete()
+    def test_pre_create_activation_key(self, activation_key_setup):
+        """Before Upgrade, Creates the activation key with different entities and update their
+        contents.
+
+        :id: preupgrade-a7443b54-eb2e-497b-8a50-92abeae01496
+
+        :steps:
+            1. Create the activation key with different entities
+            2. Add subscription in the activation key
+            3: Check the subscription id of the activation key and compare it with custom_repos
+            product id.
+            4. Update the host collection in the activation key
+
+        :parametrized: yes
+
+        :expectedresults: Activation key should be created successfully and their subscription id
+        should be same with custom repos product id.
+        """
+        ak = activation_key_setup['ak']
+        org_subscriptions = entities.Subscription(
+            organization=activation_key_setup['org']
+        ).search()
+        for subscription in org_subscriptions:
+            ak.add_subscriptions(data={'quantity': 1, 'subscription_id': subscription.id})
+        ak_subscriptions = ak.product_content()['results']
+        subscr_id = {subscr['product']['id'] for subscr in ak_subscriptions}
+        assert subscr_id == {activation_key_setup['custom_repo'].product.id}
+        ak.host_collection.append(entities.HostCollection().create())
+        ak.update(['host_collection'])
+        assert len(ak.host_collection) == 1
+
+    @post_upgrade(depend_on=test_pre_create_activation_key)
+    def test_post_crud_activation_key(self, dependent_scenario_name):
+        """After Upgrade, Activation keys entities remain the same and all their functionality works.
+
+        :id: postupgrade-a7443b54-eb2e-497b-8a50-92abeae01496
+
+        :steps:
+            1. Postupgrade, Verify activation key has same entities associated.
+            2. Update existing activation key with new entities
```

Comment:
Here there is no full stop, whereas the other lines do.

Reply:
Fixed :)

Reply:
I guess you know me by now, fell free to call me Mr Picky